plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialectDDLExtension.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.sql;
+
+/**
+ * Sql dialect ddl extension.
+ */
+public interface SQLDialectDDLExtension {
```

Comment:
Do we really need this extension interface?

I suggest adding these methods to the `SQLDialect` interface and removing this extension.

If you want to keep it, please make it extend `SQLDialect`.

Reply:
Yes, we do. DDL extension may add tens of new function. It doesn't make sense for most dialects.
Especially for NoSQL ones.